Simplenote/Classes/SPSettingsViewController+Extensions.swift

```diff
@@ -88,3 +88,55 @@ private extension BiometricAuthentication.Biometry {
         }
     }
 }
+
+// MARK: - Account Deletion
+extension SPSettingsViewController {
+
+    @objc
+    func deleteAccount() {
+        guard let user = SPAppDelegate.shared().simperium.user else {
+            return
+        }
+
+        let deletionController = accountDeletionController(for: user)
+
+        let alert = UIAlertController(title: AccountDeletion.deleteAccount,
+                                      message: AccountDeletion.confirmAlertMessage,
+                                      preferredStyle: .alert)
+        alert.addDestructiveActionWithTitle(AccountDeletion.deleteAccount) { ( _ ) in
+            deletionController.requestAccountDeletion(user)
+        }
+        alert.addCancelActionWithTitle(AccountDeletion.cancel)
+
+        present(alert, animated: true, completion: nil)
+    }
+
+    private func accountDeletionController(for user: SPUser) -> AccountDeletionController {
+        let deletionController = AccountDeletionController()
+        deletionController.successHandler = {
+            self.presentSuccessAlert(for: user)
+        }
+
+        return deletionController
```

Comment:
For consistency, we could probably move the `AccountDeletionController` initialization directly into `deleteAccount`, and move the confirmation Alert to `func persentDeletionConfirmationAlert(onConfirm: ())`

That would leave you with two `present...` API(s), and "the core business logic" handled by `deleteAccount`

Reply:
I dig it